clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java

```diff
@@ -83,8 +83,13 @@ public IamRoleCachingAgent(
 
   @Override
   public CacheResult loadData(ProviderCache providerCache) {
-    AmazonIdentityManagement iam =
-        amazonClientProvider.getIam(account, Regions.DEFAULT_REGION.getName(), false);
+    // use a region from the account for correct endpoint resolution
+    String region =
+        !account.getRegions().isEmpty()
```

Comment:
you should also check that account.getRegions.get(0) != null (account.getRegions.get(0).getName() will throw an NPE otherwise)